lib/redmine_required_field.rb

```diff
@@ -59,12 +62,30 @@ def self.related_tracker_options
     Tracker.sorted.pluck(:name, :id)
   end
 
+  def self.related_roles
+    Role.sorted.pluck(:name, :id)
+  end
+
   def self.setting_values_match_id(rrf_setting, value)
     return true if rrf_setting.blank?
     # not for ids, but may need to enable for other attribute types
     # return true if value.blank?
 
     rrf_setting.map(&:to_i).include?(value.to_i)
   end
+  def self.current_roles
+    User.current.roles.sorted.pluck(:name,:id)
+  end
+  def self.setting_values_contain_ids(rrf_setting, arr)
+    return true if rrf_setting.blank?
+    # not for ids, but may need to enable for other attribute types
+    # return true if value.blank?
+    idary = Array.new
+    arr.each do |elem|
+      idary.push(elem[1])
+    end
```

Comment:
these can be written as something like

    idary = arr.map{|k| k[1]}

Reply:
thanks a lot, i never used ruby before, so syntax is not that familiar. updated on next commit